R/Plot_synergy.R

```diff
@@ -1,328 +1,327 @@
-                       par.settings = list(axis.line=list(col="transparent")),
-                       zoom = 1, aspect = 1)
-      # 2d plot
-      layout(matrix(c(1, 2, 3, 3), nrow = 2L, ncol = 2L), heights = c(0.1, 1))
-      par(mar = c(0, 6.1, 2.1, 4.1))
-      suppressWarnings(par(mgp = c(3, -0.8, 0)))
-      plot.new()
-      plot.window(ylim = c(0, 1), xlim = range(levels), xaxs = "i", yaxs = "i")
-      rect(levels[-length(levels)],0, levels[-1L],0.3, col = col, border = NA)
-      axis(3,tick = FALSE, at = do.breaks(c(start.point, end.point), end.point/10))
-      title(plot.title)
-      par(mar = c(5.1,4.1,1.1,2.1))
-      suppressWarnings(par(mgp = c(2,1,0)))
-      plot.new()
-      mat.tmp <- t(mat.tmp)
-      x.2D <- (1:dim(mat.tmp)[1] - 1) / (dim(mat.tmp)[1] - 1)
-      y.2D <- (1:dim(mat.tmp)[2] - 1) / (dim(mat.tmp)[2] - 1)
-      plot.window(asp = NA, xlim = range(x.2D), ylim = range(y.2D), "", xaxs = "i", yaxs = "i")
-      .filled.contour(x.2D, y.2D, z = mat.tmp, levels, col = col)
-      box()
-      mtext(drug.col, 1, cex = 1, padj = 3)
-      mtext(drug.row, 2, cex = 1, las = 3, padj = -3)
-      if(!is.null(row.range)) {
-        yconc <- round(row.conc[row.range[1]:row.range[2]], 3)
-      } else {
-        yconc <- round(row.conc, 3)
-      }
-      
-      if(!is.null(col.range)) {
-        xconc <- round(col.conc[col.range[1]:col.range[2]], 3)
-      } else {
-        xconc <- round(col.conc, 3)
-      }
-      
-      axis(side = 1, at = seq(0, 1, by = 1/(length(xconc) - 1)), labels = xconc)
-      axis(side = 2, at = seq(0, 1, by = 1/(length(yconc) - 1)), labels = yconc)
-      print(syn.3d.plot, position = c(0.5,0, 1, 1), newpage = FALSE)
-      fig <- recordPlot()
-    }
-    plots[[i]] <- fig
-    if(save.file) {
-      if (type %in% c("2D", "3D")) {
-        pdf(file.name, width = 10, height = 10)
-      } else {
-        pdf(file.name, width = 12, height = 6)
-      }
-      print(fig)
-      dev.off()
-    }
+# Returns nested associative lists with all data needed for plots of any type.
+.GetPlotData <- function(
+  data, index, legend.start, legend.end, row.range, col.range, len) {
+  scores.dose <- data$scores[[index]]
+  color.range <- .RoundUpToNextTen(scores.dose)
+  unit.text <- paste("(", data$drug.pairs$concUnit[index], ")", sep = "")
+  return(list(
+    metadata = list(  # useful for naming output files
+      drug.blockID = data$drug.pairs$blockIDs[index],
+      file.name.prefix = paste(data$drug.pairs$drug.row[index],
+                               data$drug.pairs$drug.col[index], sep = '.')
+    ),
+    data = list(  # useful for actual plot content
+      plot.title =
+        .ComputePlotTitle(scores.dose, row.range, col.range, data$method),
+      # Note, when considering application of ranges, that with len=3 mat.tmp will
+      # have 9 times as many elements as scores.dose.  It is best to invoke
+      # .ExtendedScores only after ranges have been applied.
```

Comment:
Agreed. Previously, I used Kriging package for the interpolation which was not working well with small matrixes. That's why I extend the score matrix first and then slice the matrix based on the selected ranges. I changed to SpatialExtremes package for the interpolation which seems to be good at very small matrixes as well. So I think applying ranges first and then extending the scores are good.

Reply:
Acknowledged.